Add indictor and timestamp labels

notlee opened the pull request:
Addresses: https://github.com/Financial-Times/o-labels/issues/80
Beta release in the registry: https://registry.origami.ft.com/components/o-labels@5.1.0-beta.1.0.0/

Adds indicator and timestamp labels to `o-labels` with an inverse variant. These styles 
match  styles found in [o-teaser](https://github.com/Financial-Times/o-teaser/blob/master/src/scss/elements/_timestamp.scss) and [live blogs](https://github.com/Financial-Times/next-article/blob/cf51a94382cb9fb1b98dfb144781e456565736ca/client/components/live-event/_status.scss#L1), which will be updated to remove 
duplication by using `o-labels`.

If a `font-size`, `line-height`, or `color` property is set on the root element these new 
indicator styles will adapt accordingly. This will allow us to add new variants in the future 
with minimal extra styles or enable products to overrides these properties for one-off uses 
more easily.

<img width="1428" alt="Screenshot 2020-07-13 at 14 25 48" src="https://user-images.githubusercontent.com/10405691/87309657-cc8c8380-c514-11ea-9e1f-670b9f343786.png">
<img width="1428" alt="Screenshot 2020-07-13 at 14 25 52" src="https://user-images.githubusercontent.com/10405691/87309662-cdbdb080-c514-11ea-90ea-e041ce492113.png">
<img width="1428" alt="Screenshot 2020-07-13 at 14 25 54" src="https://user-images.githubusercontent.com/10405691/87309667-ceeedd80-c514-11ea-8469-a3a61bc97d27.png">
<img width="1428" alt="Screenshot 2020-07-13 at 14 25 56" src="https://user-images.githubusercontent.com/10405691/87309673-d0200a80-c514-11ea-9d67-df942b17f2a8.png">

New "content" mixins will allow us to use indicator and timestamp labels in other
components (i.e. [o-teaser](https://github.com/Financial-Times/o-teaser/blob/master/src/scss/elements/_timestamp.scss)).
As the indicator labels use multiple html elements the mixin needs to be able to output
styles for each element separately. This is a little complicated. Difficult to document and
maintain. It links back to the [component composition issue](https://github.com/Financial-Times/origami/issues/45).

Commit:
wip add indicator and timestamp styles

Commit:
document new label markup

Commit:
Add docs for new indicator label sass to readme

notlee commented:
**do not release: let's release a beta to get feedback from content innovation and test out the o-teaser integration first**

origamiserviceuser commented:
o-labels bundle size difference from 5.0.9 to 10305e69bc1367bcd18630f99ce4e6385a09fe5d
css, master: 1.73kb increase (0.37kb/gzip)
css, internal: 1.72kb increase (0.38kb/gzip)
css, whitelabel: 1.58kb increase (0.36kb/gzip)

JakeChampion commented on README.md:
```diff
@@ -1,24 +1,84 @@
+## Label Types
+
+There are three types of label:
+- A standard label, the default.
+- An indicator label.
+- A timestamp label.
```
What do we think about making these label types link to their documentation?
```suggestion
- [A standard label](#standard-label), the default.
- [An indicator label](#indicator-label).
- [A timestamp label](#timestamp-label).
```

JakeChampion commented on bower.json:
```diff
@@ -7,6 +7,7 @@
     "o-brand": "^3.2.5",
-    "o-spacing": "^2.0.0"
+    "o-spacing": "^2.0.0",
+    "o-editorial-typography": "^1.0.10"
```
Does this make it a breaking change?

notlee replied:
thankfully not as it's at v1 and is compatible with the latest major of all other components

Commit:
only output indicator/timestamp css with brand support

notlee committed:
Update README.md

Co-authored-by: Jake Champion <me@jakechampion.name>

origamiserviceuser commented:
o-labels bundle size difference from 5.0.9 to 44425c25045877ba29602171253934cbf7f2cb75
css, master: 1.73kb increase (0.37kb/gzip)
An insignificant difference was also found for: css (internal), css (whitelabel)

origamiserviceuser commented:
o-labels bundle size difference from 5.0.9 to 7c501c2395ba8dc3d28e804f5719ba2c8c828ad1
css, master: 1.73kb increase (0.37kb/gzip)
An insignificant difference was also found for: css (internal), css (whitelabel)

notlee commented:
thanks to fox friend I noticed this was outputting styles for brands it shouldn't be :)

Commit:
Update demo css so live indicator overflow is not hidden

origamiserviceuser commented:
o-labels bundle size difference from 5.0.9 to 9b3f4fe6ba087ebd0a4473feddbd5b28c1c75f44
css, master: 1.73kb increase (0.37kb/gzip)
An insignificant difference was also found for: css (internal), css (whitelabel)

notlee commented:
here's the beta release in the registry: https://registry.origami.ft.com/components/o-labels@5.1.0-beta.1.0.0/

notlee commented:
note: do we want to stop the live label animation if `prefers-reduced-motion` is set?
https://financialtimes.slack.com/archives/C2LMEKC6S/p1594654150152900

Commit:
Prevent the live label animation if `prefers-reduced-motion` is set.

Prevent animation of the live label if the reader has indicated
that they prefer reduced motion. This may be conservative but
as the live indicator is presented alongside content (including
the front page at the time of writing) we do not want to risk
severely distracting some readers.
https://www.w3.org/WAI/WCAG21/Understanding/pause-stop-hide.html

origamiserviceuser commented:
o-labels bundle size difference from 5.0.9 to 0a18b9538b42ba3bb7c04f2d517113dfa9fbd1d9
css, master: 1.85kb increase (0.40kb/gzip)
An insignificant difference was also found for: css (internal), css (whitelabel)

ker-an commented on README.md:
```diff
@@ -63,71 +119,129 @@ The following states are used to represent the FT's service tiers:
+Indicator labels have one of three statuses:
+- `live`
+- `updated`
+- `new`
```
For live blogs, we also have a `closed` status which makes it clear that an old live blog is closed. Would it be possible to add this status? Otherwise, we'll need to add a different attribute to the markup and style it separately in `next-article`. We use the `live` and `closed` statuses the most in live blogs.

ker-an commented on README.md:
```diff
@@ -63,71 +119,129 @@ The following states are used to represent the FT's service tiers:
 
-### Mixin: `oLabelsContent`
+When it's not possible to use an `o-labels` CSS class, for example within another Origami component, use:
+- `oLabelsContent` to output a standard label with a custom class.
```
OH! Can we use `oLabelsContent` to create our own `closed` label?

ker-an commented on README.md:
```diff
@@ -63,71 +119,129 @@ The following states are used to represent the FT's service tiers:
-### Mixin: `oLabelsContent`
+When it's not possible to use an `o-labels` CSS class, for example within another Origami component, use:
+- `oLabelsContent` to output a standard label with a custom class.
+- `oLabelsIndicatorContent` to output an indicator label with a custom class.
```
For live blogs, we also have a `closed` status which makes it clear that an old live blog is closed. Would it be possible to add this status? Otherwise, we'll need to add a different attribute to the markup and style it separately in `next-article`. We use the `live` and `closed` statuses the most in live blogs.

OR can we use `oLabelsIndicatorContent` to create our own `closed` indicator?

notlee replied:
thanks for flagging I forgot about `closed`! :) It might be a little preemptive given its only used in live blogs at the moment but I think I'll add a `closed` variant. It'll allow us the flexibility of changing the style of the closed status independently of other statuses and allow us to avoid using the `oLabelsIndicatorContent` api, which is a bit cumbersome.

I've seen references to a `comingsoon` label too. After chatting with Josh we figured `comingsoon` is not used and I haven't included it. Do you think that's right?

ker-an replied:
Yup, we don't use the `comingsoon` label.

Commit:
Add `closed` indicator label

origamiserviceuser commented:
o-labels bundle size difference from 5.0.9 to 7cb5a80ea96601a197d348f22fba1b35e8a82dad
css, master: 1.92kb increase (0.41kb/gzip)
An insignificant difference was also found for: css (internal), css (whitelabel)

JakeChampion commented on README.md:
```diff
@@ -59,6 +59,7 @@ This table outlines the possible indicator label statuses:
 | Size                 | Description                                                   | Brand support |
 |----------------------|---------------------------------------------------------------|---------------|
 | live                 | Indicate a story is live.                                     | master        |
+| live                 | Indicate a live story is no longer live.                                     | master        |
```
```suggestion
| closed                 | Indicate a live story is no longer live.                                     | master        |
```

Commit:
correct readme

origamiserviceuser commented:
o-labels bundle size difference from 5.0.9 to 31852d2419f1e82fec298893aa28acabfc13989f
css, master: 1.92kb increase (0.41kb/gzip)
An insignificant difference was also found for: css (internal), css (whitelabel)

notlee commented:
Closed label added! Anything else you'd like to add before release @ker-an? :)
https://registry.origami.ft.com/components/o-labels@5.1.0-beta.1.1.0/#demo-indicators

ker-an reviewed (approved):
Nope, nothing more to add from me! Thank you, this looks great.

origamiserviceuser commented:
:tada: This PR is included in version v5.1.0 :tada:
